system/ee/legacy/controllers/cp/login.php

```diff
@@ -795,6 +799,31 @@ private function _return_to_login($lang_key)
 
         $this->functions->redirect(BASE . AMP . $redirect);
     }
+
+    /**
+     * AJAX endpoint for password strength meter
+     *
+     * @return JSON
+     */
+    public function validate_password()
```

Comment:
@intoeetive can we use the validation service here instead of having this function here?

Reply:
This is the controller function providing the AJAX endpoint to show password strength meter. It is not possible to use validation service directly as a controller - and also the validation service can only return boolean value (valid or not valid), not the rank number/text